### DISCUSSION TITLE
Add withCasts to Query Builder

### DISCUSSION THREAD

**Discussion by euoia** [laravel/framework#49367] [Ideas]
Pretty much the proposal made here:
https://github.com/laravel/ideas/issues/2134

`withCasts` is available on Eloquent queries, but is not available on the query builder. This would be useful, for example when making JSON columns with DB::raw (e.g. MySQL's JSON_ARRAYAGG).

I can work around it by forcing the query through Eloquent's query builder, but this seems less than ideal (not least because I need to add `withTrashed()` to prevent query builder adding a `WHERE deleted_at IS NULL` - even though I am not selected `deleted_at` in my query).

**Comment by vibonacci** [comment]
This would be extremely useful for full custom queries that skip Eloquent, yet still want to use casting features of Laravel like created_at -> datetime.
